test/type_failures.ml

```diff
@@ -56,3 +56,9 @@ let%expect_test "Cannot reassign int to something stored with more bits" =
 let%expect_test "Non-existent type" =
   compile_string_with_failure "func foo(a: non_type[10]) { let x = 2; }";
   [%expect {| [Type Error] No type definition for `non_type'.; |}]
+
```

Comment:
Can you add the following tests:
- The view is created from a non-array.
- The view is created using an iterator that is being unrolled.